projects/packages/blaze/src/class-blaze.php

```diff
@@ -214,12 +214,20 @@ public static function jetpack_blaze_row_action( $post_actions, $post ) {
 	 * Enqueue block editor assets.
 	 */
 	public static function enqueue_block_editor_assets() {
+		// Bail if criteria is not met to enable Blaze features.
+		if ( ! self::should_initialize() ) {
+			return;
+		}
```

Comment:
Personally, I'd leave this where it was. The check for the current screen looks cheaper than the connection check, so better to do that first. They're still both being done before any other checks.

Reply:
that's fair. I'll switch them back around. Having the `get_current_screen` first will prevent the fatal, which is the ultimate initial goal. it fataling for a disconnected site got under my skin, but you're right that if the fatal is covered, it doesn't matter.

Reply:
[f3a8613](https://github.com/Automattic/jetpack/pull/28955/commits/f3a8613462181292a96fcbcd1cf527f24b81f7de)